r/src/arrow_types.h

```diff
@@ -38,6 +38,8 @@
 #include <arrow/dataset/type_fwd.h>
 #endif
 
+#include <arrow/compute/exec/options.h>
```

Comment:
This was moved over here so that `compute::JoinType` can be passed from R instead of having to pass an integer (and then converting within C++).